python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -425,6 +425,13 @@ def load_input(self, context: InputContext) -> Union[Any, Dict[str, Any]]:
 
                 return self._load_multiple_inputs(context)
 
+    def _handle_transition_to_partitioned_asset(self, path: "UPath"):
+        # if the asset was previously non-partitioned, path will be a file, when it should
+        # be a directory for the partitioned asset. Delete the file, so that it can be made
+        # into a directory
+        if path.exists() and path.is_file():
+            path.unlink()
```

Comment:
* I think we should `context.log.warn` when this happens for clarity to user
* do we need to worry about this happening concurrently? Upath io manager has that coroutine support for partitions for somethings (i forget if its only load input or also handle output). I think you can do `missing_ok=True` to make it so this is safe if theres a race and we try to unlink something that is gone

Reply:
I'll add the `missing_ok=True` just to be safe. two materializations of the same asset that both detect the existing file, but one deletes it before the other tries to delete is unlikely but possible.